internal/store/root/move.go

```diff
@@ -156,6 +147,43 @@ func (r *Store) moveFromTo(ctx context.Context, subFrom *leaf.Store, from, to, f
 	return nil
 }
 
+func computeMoveDestination(src, from, to string, srcIsDir, dstIsDir bool) string {
+	// special case: moving up to the root
+	if to == "." || to == "/" {
+		dstIsDir = false
+		to = ""
+	}
+
+	// are we moving into an existing directory? Then we just need to prepend
+	// it's name to the source.
+	// a -> b
+	// - a/f1 -> b/a/f1
+	// a -> b
+	// - a -> b/a
```

Comment:
Wow, I've no idea what that means.

`mv a b` on a tree with a/f1 inside of a?
And same but for just an entry named a?